### ISSUE TITLE
Crash opening camera on Huawei Mediapad T5

### ISSUE THREAD

**Issue by tommag21** [NativeScript/nativescript-camera#243]
### Which platform(s) does your issue occur on?
- Huawei Mediapad T5 (AGS2-L09), EMUI 8.0.0, Android 8.0.0

### Please, provide the following version numbers that your issue occurs with:

The NS-Vue demo of this plugin and this app:

- CLI: 6.4.1
- Cross-platform modules: 6.0.0, 6.5.1
- Runtime(s): 6.4.1, 6.5.0
- Plugin(s):
"dependencies": {
        "@nativescript/theme": "^2.2.1",
        "@nstudio/nativescript-checkbox": "^1.0.0",
        "axios": "^0.19.0",
        "crypto-js": "^4.0.0",
        "hi-base32": "^0.5.0",
        "md5": "^2.2.1",
        "moment": "^2.24.0",
        "nativescript-bitmap-factory": "^1.8.1",
        "nativescript-bluetooth": "^3.0.0-beta.10",
        "nativescript-camera": "^4.5.0",
        "nativescript-drawingpad": "^3.1.0",
        "nativescript-feedback": "^1.3.12",
        "nativescript-fonticon": "^2.0.0",
        "nativescript-geolocation": "^5.1.0",
        "nativescript-iqkeyboardmanager": "^1.5.1",
        "nativescript-localize": "^4.2.0",
        "nativescript-modal-datetimepicker": "^1.2.1",
        "nativescript-orientation": "^2.2.4",
        "nativescript-print-zebra": "file:nativescript-print-zebra-1.2.0.tgz",
        "nativescript-printer": "^1.5.0",
        "nativescript-sqlite": "^2.6.0",
        "nativescript-sqlite-commercial": "file:nativescript-sqlite-commercial-1.3.2.tgz",
        "nativescript-sqlite-encrypted": "file:nativescript-sqlite-encrypted-1.4.0.tgz",
        "nativescript-toasty": "^3.0.0-alpha.2",
        "nativescript-ui-autocomplete": "^6.0.0",
        "nativescript-ui-listview": "^8.0.1",
        "nativescript-vue": "^2.5.0",
        "nativescript-zip": "^4.0.2",
        "pdfmake": "^0.1.63",
        "tns-core-modules": "^6.5.1",
        "vuex": "^3.1.2",
        "xml-js": "^1.6.11"
},
"devDependencies": {
        "@babel/core": "^7.0.0",
        "@babel/preset-env": "^7.0.0",
        "babel-loader": "^8.0.2",
        "eslint": "^6.7.2",
        "eslint-plugin-vue": "^6.2.2",
        "nativescript-dev-webpack": "^1.3.0",
        "nativescript-vue-template-compiler": "^2.5.0",
        "node-sass": "^4.13.1",
        "vue-loader": "^15.9.0"
}

### Please, tell us how to recreate the issue in as much detail as possible. 
When I open the camera, the NativeScript activity restarts in the background, showing the splash page and the first page.
On the demo app it happens about 15% of the times, on my app about 90%.

Debug stops [here](https://github.com/NativeScript/nativescript-camera/blob/master/src/camera.android.ts#L151) and [this](https://github.com/NativeScript/nativescript-camera/blob/master/src/camera.android.ts#L90) event handler is never called.
The plugin works fine on other devices and emulators.

Full device logs:
[output.txt](https://github.com/NativeScript/nativescript-camera/files/4475834/output.txt)

### Is there any code involved? 
```js
async addIdDocument() {
    try {
        await camera.requestCameraPermissions();
        let asset = await camera.takePicture({ width: 1024, height: 1024, keepAspectRatio: true }); // same result without options parameter
        this.idDocument = await ImageSource.fromAsset(asset);
    } catch (e) {
        if (e.message === "cancelled") return;
        this.$snackbar(this.$options.filters.L("newRecupero.fotocameraError"), "error");
        console.log(e);
    }
}
```

**Comment by RGBvision** [comment]
Same issue on various Android devices (Android 6 - 9). One or two seconds after photo has been taken and appears on Image component App returns to start page.

**Comment by tommag21** [comment]
@RGBvision what NativeScript version are you using?

**Comment by RGBvision** [comment]
@tommag21 tns-core-modules 6.5.1, tns-android 6.5.0

**Comment by RGBvision** [comment]
I found that when you open the camera and don't take a photo, just press back button, the entire history of navigation is lost. Start page will be open. Looks like app restarting anyway.

**Comment by tommag21** [comment]
I am experiencing the same bug on the demo-vue app, plus another one which might be related.

The more photos I take, the slower the app gets, until it crashes. It takes about 4-5 snaps.

Logs:
[debug-demo.txt](https://github.com/NativeScript/nativescript-camera/files/4517074/debug-demo.txt)

> I found that when you open the camera and don't take a photo, just press back button, the entire history of navigation is lost. Start page will be open. Looks like app restarting anyway.

From what I've seen the crash happens when the camera is opened, before the photo is taken.

**Comment by tommag21** [comment]
> I found that when you open the camera and don't take a photo, just press back button, the entire history of navigation is lost. Start page will be open. Looks like app restarting anyway.

From what I've seen the crash happens when the camera is opened, before the photo is taken.

**Comment by tommag21** [comment]
Might be related to this:
[White screen, no crash, logged ClassNotFoundException OnUnhandledKeyEventListener](https://github.com/NativeScript/android-runtime/issues/1484)

**Comment by tommag21** [comment]
After further investigation I have concluded that it's a low memory issue. On some devices like the Huawei Mediapad T5, opening the camera causes the sistem to sometimes kill background apps, including the one launching the camera intent.
The taken photo is still saved at [these paths](https://github.com/NativeScript/nativescript-camera/blob/master/src/camera.android.ts#L54), so I edited the plugin code to save the path to [application settings](https://docs.nativescript.org/ns-framework-modules/application-settings), retrieving the photo after the app has restarted.
